### ISSUE TITLE
Do you have the code or details on how this used to work with PDF extraction?

### ISSUE THREAD

**Issue by TercioOfParma** [rotula/simplestyle#2]
I am working on a project where I am trying to modify PDFMiner to extract subscripts and superscripts and I would really appreciate your input on this. 

Thanks a lot in advance.

**Comment by rotula** [comment]
Our PDF extraction also works with PDFMiner. I forked the original PDFMiner, but later on merged [PDFMiner.six](https://github.com/pdfminer/pdfminer.six) into it. Our project is here: [https://github.com/rotula/pdfminer](https://github.com/rotula/pdfminer) I changed a lot of the original output routines so I am no longer sure, which parts of the character information is added by my fork and which parts were present in the original. Your best bet might be to checkout our project [anapdf](https://github.com/rotula/anapdf), which uses our PDFMiner fork and interprets the rise- and origin-attributes added to the character output in order to determine subscript or superscript.
Note that our PDFMiner and anapdf projects are very much work in progress and very much geared towards our specific needs (lots of text, not many images), so your mileage may vary considerably.
As this is not actually an issue with simplestyle, I'll close this issue, but feel free to contact me directly via email: clemens.radl@googlemail.com

**Comment by TercioOfParma** [comment]
Thanks a lot. I didn't know precisely where to ask so I just posted here.

**Comment by rotula** [comment]
OK, I've had a chance to take a closer look at the code. I tweaked LTChar to have a [property "origin"](https://github.com/rotula/pdfminer/blob/e7e9a0ed8e5ad87db5d39dc3dad406505d23b49c/pdfminer/layout.py#L243), but this seems to be just the translation part of the matrix which is another property of LTChar. So if you somehow get hold of the LTChars in your code, you are set. Our programs work with the XML output of PDFMiner, so I added the properties to the [appropriate tag](https://github.com/rotula/pdfminer/blob/e7e9a0ed8e5ad87db5d39dc3dad406505d23b49c/pdfminer/converter.py#L529).
And in our later programs I try to [establish a baseline for each textline](https://github.com/rotula/anapdf/blob/13d838fa96cab35e82720ae3feeab836908d1875/src/anapdf/converters.py#L322) (which is easy in our case, as we do not work with formulas but just normal text with occasional footnotes). And finally, I re-iterate through the characters of each line and [compare the rise-attribute or the origin with the baseline of the current textline](https://github.com/rotula/anapdf/blob/13d838fa96cab35e82720ae3feeab836908d1875/src/anapdf/converters.py#L444). If a certain threshold is met, the character gets the superscript or subscript property.
